HCCGo/app/js/jobViewCtrl.js

```diff
@@ -75,4 +95,71 @@ jobViewModule.controller('jobViewCtrl', ['$scope', '$log', '$timeout', 'connecti
     });
   });
 
+  /**
+   * Saves the output or error to a file
+   * @method saveFile
+   * @memberof HCCGo.jobViewCtrl
+   * @param {String} fileType - Denotes either the output or error to be Saved
+   * @param {Event} $event - Used to stop the save event from moving up the DOM
+   */
+  $scope.saveFile = function(fileType, $event) {
+    const {dialog} = require('electron').remote;
+    const fs = require('fs');
+    const path = require('path');
+
+    var options = {
+      filters: [
+        { name: 'text', extensions: ['txt']}
+      ]
+    };
+
+    dialog.showSaveDialog(options, function(localFile) {
+      // If the user clicks cancel, localFile will be undefined
+      if (!localFile) return;
+
+      if (!downloadRemoteFile($scope.job[fileType], localFile)) {
+        fs.writeFile(localFile, $scope.job[fileType], function(err) {});
+      }
+      notifierService.success(path.basename(localFile).trim() + " Was Successfully Saved!", "Save Successful!");
+    });
+    // Stop the propagation of the click
+    $event.stopPropagation();
+  };
+
+  /**
+   * Used to download the file requested from the server, since only 5MB worth of data is displayed to the user.
+   * @method downloadRemoteFile
+   * @memberof HCCGo.jobViewCtrl
+   * @param {String} remotePath - Denotes the path on the server to download the file from
+   * @param {String} localPath - Denotes the path where the user wants the file saved
+   * @returns {Boolean} A flag that denotes whether the file needed to be downloaded from the server or not
+   */
+  function downloadRemoteFile(remotePath, localPath) {
+    connectionService.getFileSize(remotePath).then(function(size) {
+      if (size > 5*1025*1024) {
+        connectionService.quickDownload(remotePath, localPath).then(function(msg) {
+          if (msg !== null){
+            notifierService.error(msg);
+            return false;
+          }
+        });
+        return true;
```

Comment:
Again, where is this return going?  It's inside a `then`, so... no where?  You have to use promises here.